bin/ck8s

```diff
@@ -83,6 +85,11 @@ upgrade)
   export CK8S_CLUSTER="${2}"
   "${here}/upgrade.bash" "${3}" "${4}"
   ;;
+upgrade-unlock)
+  [[ "${2}" =~ ^(wc|sc|both)$ ]] || usage
+  export CK8S_CLUSTER="${2}"
+  "${here}/upgrade.bash" "unlock"
+  ;;
```

Comment:
```bash
  echo "  upgrade <wc|sc|both> unlock                     reset a migration to allow trying again" 1>&2
####
upgrade)
  [[ "${2}" =~ ^(wc|sc|both)$ ]] || usage
  if [[ "${3}" == "unlock" ]]; then
    export CK8S_CLUSTER="${2}"
    "${here}/upgrade.bash" "unlock"
    exit 0
  fi
  [[ "${3}" =~ ^(v[0-9]+\.[0-9]+)$ ]] || usage
  [[ "${4}" =~ ^(prepare|apply)$ ]] || usage
  check_tools
  export CK8S_CLUSTER="${2}"
  "${here}/upgrade.bash" "${3}" "${4}"
  ;;
```

I guess this would work, tho feels a bit awkward to jump out in the middle here. @aarnq ?

Reply:
Depends on how you write it, you don't have to jump in the middle.

```sh
[[ "${2}" =~ ^(wc|sc|both)$ ]] || usage
[[ "${3}" =~ ^(v[0-9]+\.[0-9]+|unlock)$ ]] || usage
if [[ "${3}" != "unlock" ]]; then
  [[ "${4}" =~ ^(prepare|apply)$ ]] || usage
fi
check_tools
export CK8S_CLUSTER="${2}"
"${here}/upgrade.bash" "${3}" "${4:-}"
```

Reply:
Is this resolved?

Reply:
This was about whether the `unlock` command should be part of `upgrade`, i.e. `upgrade <wc|sc|both> prepare` vs `upgrade-unlock <wc|sc|both>` and how to handle the different number of in-the-middle arguments.